src/DIRAC/DataManagementSystem/scripts/dirac_dms_directory_sync.py

```diff
@@ -29,480 +29,497 @@
+        pro = Process(target=function, args=argums)
+        pro.start()
+        processes.append(pro)
+    for process in processes:
+        process.join()
+
+    if any(process.exitcode == 1 for process in processes):
+        return S_ERROR()
+    return S_OK()
+
+
+def syncDestinations(upload, source_dir, dest_dir, storage, delete, nthreads):
+    """
+    Top level wrapper to execute functions
+    """
+    from DIRAC.Resources.Catalog.FileCatalog import FileCatalog
+    from DIRAC.DataManagementSystem.Client.DataManager import DataManager
+
+    fc = FileCatalog()
+    dm = DataManager()
+
+    result = getContentToSync(upload, fc, source_dir, dest_dir)
+    if not result["OK"]:
+        return S_ERROR(result["Message"])
+
+    if upload:
+        res = doUpload(fc, dm, result, source_dir, dest_dir, storage, delete, nthreads)
         if not res["OK"]:
-            return S_ERROR(res["Message"])
+            return S_ERROR("Upload failed: " + res["Message"])
+    else:
+        res = doDownload(dm, result, source_dir, dest_dir, delete, nthreads)
+        if not res["OK"]:
+            return S_ERROR("Download failed: " + res["Message"])
+
+    return S_OK("Mirroring successfully finished")
+
+
+def run(parameters, delete, nthreads):
+    """
+    The main user interface
+    """
+
+    source_dir = parameters[0]
+    dest_dir = parameters[1]
+    upload = False
+    storage = None
+
+    if len(parameters) == 3:
+        storage = parameters[2]
+        source_dir = os.path.abspath(source_dir)
+        dest_dir = dest_dir.rstrip("/")
+        upload = True
+        if not os.path.isdir(source_dir):
+            gLogger.fatal("Source directory does not exist")
+            DIRAC.exit(1)
 
-        return S_OK("Successfully mirrored " + source_dir + " into " + dest_dir)
+    if len(parameters) == 2:
+        dest_dir = os.path.abspath(dest_dir)
+        source_dir = source_dir.rstrip("/")
+        if not os.path.isdir(dest_dir):
+            gLogger.fatal("Destination directory does not exist")
+            DIRAC.exit(1)
+
+    res = syncDestinations(upload, source_dir, dest_dir, storage, delete, nthreads)
+    if not res["OK"]:
+        return S_ERROR(res["Message"])
+
+    return S_OK("Successfully mirrored " + source_dir + " into " + dest_dir)
+
+
+@DIRACScript()
+def main():
+    Script.registerSwitch("D", "sync", "Make target directory identical to source")
+    Script.registerSwitch("j:", "parallel=", "Multithreaded download and upload")
+    Script.parseCommandLine(ignoreErrors=False)
+
+    args = Script.getPositionalArgs()
+    if len(args) < 1 or len(args) > 3:
+        Script.showHelp()
+
+    sync = False
+    parallel = 1
+    for switch in Script.getUnprocessedSwitches():
+        if switch[0].lower() == "s" or switch[0].lower() == "sync":
+            sync = True
+        if switch[0].lower() == "j" or switch[0].lower() == "parallel":
+            parallel = int(switch[1])
+
+    listOfFailedFiles = Manager().list()
```

Comment:
This variable was used as a "global" variable before. I don't think this is going to work after these changes.

Reply:
Yeah, this is what pylint says as well